server/metrics/metrics.go

```diff
@@ -109,3 +119,35 @@ func (m *MetricsServer) IncExtensionRequestCounter(extension string, status int)
 func (m *MetricsServer) ObserveExtensionRequestDuration(extension string, duration time.Duration) {
 	m.extensionRequestDuration.WithLabelValues(extension).Observe(duration.Seconds())
 }
+
+
+func InitDAUMetrics(duration time.Duration) {
+	DAUMetricsInstance = &DAUMetrics{
+		userSet:  make(map[string]time.Time),
+		duration: duration,
+		gauge: prometheus.NewGauge(prometheus.GaugeOpts{
```

Comment:
Would it be easier to use a counter? Every time a user logs in, increment the counter. you can then use promql and other tools to get the delta between any time period.

Reply:
Using a gauge seems more appropriate for this use case. DAU (Daily Active Users) focuses on tracking unique users over a 24-hour period. I believe a gauge is the better choice!

Reply:
What advantages does a gauge provide

Reply:
I initially thought it would be necessary to measure the number of active users using a gauge, as it could go up and down depending on user activity. However, considering the requirement to track daily active users as a continually increasing count, a counter is more appropriate. I will adjust the implementation accordingly to use a counter.